include/sys/trace.h

```diff
@@ -43,19 +43,23 @@
  */
 #undef DTRACE_PROBE1
 #define	DTRACE_PROBE1(name, t1, arg1) \
-	trace_zfs_##name((arg1))
+	if (arg1) \
+		trace_zfs_##name((arg1))
 
 #undef DTRACE_PROBE2
 #define	DTRACE_PROBE2(name, t1, arg1, t2, arg2) \
-	trace_zfs_##name((arg1), (arg2))
+	if (arg1 && arg2) \
```

Comment:
@loli10K how did u generate the above output...Are the NULL entries manually updated to denote the expected results? Talking about Linux tracepoints, I have not encountered any instance in Linux code where the args passed to the tracing function are NULL. Moreover, based on my understanding of Linux tracepoints and the API it exposes, there is no way to handle NULL without making changing to the tracing functions. As far as i know its the responsibility of the caller to make sure args are non-null just as any other Linux logging infra (printk etc)

With respect to the new patch, the intent is to have the developer not worry about NULL args, otherwise, i would follow the same semantics followed in other places which is to trace only with Non-NULL args.

But if still, we want the results to look like you mentioned, we can probably, create a dummy tracepoint, that just prints NULL.

Let me know what you think and accordingly i can modify the patch.